tests/conftest.py

```diff
@@ -72,11 +74,44 @@ def compile_and_run(device, reset_torch_dynamo, request):
             accuracy = calculate_accuracy(outputs, outputs_after)
             if accuracy:
                 comp_runtime_metrics["accuracy"] = accuracy
+
+            # Memory analysis
+            mm = option._memory_manager
+            compiled_memory_metric = {}
+            peak_usage = mm.peak_sram_usage / 1048576
+            peak_sram_usage_metric = {"peak_sram_usage": peak_usage}
+
+            if mem_utils.check_sram_overflow(mm) is True:
+                compiled_memory_metric = {"fits_in_memory": "No"}
+            else:
+                compiled_memory_metric = {"fits_in_memory": "Yes"}
+
+            # These are for plotting charts for later inspection
+            from tools.plot_chart import plot_bar_chart, plot_line_chart
+
+            bar_chart_file = f"metrics/{model_name}/bar_chart.png"
+            line_chart_file = f"metrics/{model_name}/line_chart.png"
+            plot_bar_chart(mm.data_points, bar_chart_file)
+            plot_line_chart(mm.data_points, line_chart_file)
+
+            log_file = f"metrics/{model_name}/memory_footprint.txt"
+            with open(log_file, "w") as f:
+                f.write(mm.logs)
+
         except Exception as e:
             comp_runtime_metrics = {"success": False}
+            compiled_memory_metric = {"fits_in_memory": "N/A"}
+            peak_sram_usage_metric = {"peak_sram_usage": 0}
             print(f"{model_name} compiled failed to run. Raised exception: {e}")
             raise
+
         finally:
             compiled_metrics_path = p / f"compiled-run_time_metrics.pickle"
             with open(compiled_metrics_path, "wb") as f:
                 pickle.dump(comp_runtime_metrics, f)
+            compiled_memory_metrics_path = p / f"compiled_memory_metrics.pickle"
+            with open(compiled_memory_metrics_path, "wb") as f:
+                pickle.dump(compiled_memory_metric, f)
+            peak_sram_usage_metric_path = p / f"peak_sram_usage_metrics.pickle"
+            with open(peak_sram_usage_metric_path, "wb") as f:
+                pickle.dump(peak_sram_usage_metric, f)
```

Comment:
I think this has to be merged into the compuled-run_time_metrics.
Please align with @kevinwuTT and if so, please merge into

Reply:
Sure, since memory analysis is done during the compile run, we can put the metrics into the same file so it doesn't get too messy. And in the `collect_metrics.py` just check if the key exists for the memory metrics from the same file.

Reply:
I have done that.